### DISCUSSION TITLE
How to fetchNextPage of infiniteQuery using queryClient?

### DISCUSSION THREAD

**Discussion by afiiif** [TanStack/query#1958] [Q&A] [answered]
I have an infiniteQuery, say like this,
```javascript
const queryNotifications = useInfiniteQuery('notifications', getNotifications, {
  getNextPageParam: someFunction,
});
```

Then, I have another component, which use `useQueryClient` hook.  
I want to fetchNextPage from that component.

Is it possible?  
```javascript
const queryClient = useQueryClient();
queryClient.fetchNextPage('notifications'); // Something like this
```
How can I do fetchNextPage from this component?

**Comment by TkDodo** [comment, marked as answer]
I don’t think it’s possible with just the queryClient. You can either useInfiniteQuery in the second component as well, or take the function from the first component and make it available to the second one (as a prop or via context or so)

**Reply by afiiif** [reply]
Thank you...
